### PR TITLE
tests

### PR DESCRIPTION
**Pull request by StalkerSanya**
added new package with unit test and integration test

### CODE REVIEW

File: my_robot_gazebo/launch/xacro.launch [hunk 1]
```diff
@@ -8,28 +8,28 @@
...
-										joint8_position_controller"/>
+										joint1_position_controller"/> -->
 									<!--joint2_position_controller
 										joint3_position_controller
 										joint4_position_controller
 										joint5_position_controller
 										joint6_position_controller
 										joint7_position_controller
-										joint1_position_controller"/> -->
+										joint8_position_controller"/> -->
```
Comment (l1va) [inline my_robot_gazebo/launch/xacro.launch]: why do you hide it?

File: tests/integration_test/scripts/sub.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		print("failed")
+
+def check_joint7(msg):
+	if abs(msg.set_point-msg.process_value) < 0.01:
+		print('joint 7 passed test')
+	else:
+		print("failed")
+
+def check_joint8(msg):
+	if abs(msg.set_point-msg.process_value) < 0.01:
```
Comment (l1va) [inline tests/integration_test/scripts/sub.py]: where are asserts?

File: tests/integration_test/scripts/sub.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	else:
+		print("failed")
+
+def my_robot_joint_positions_subscriber():
+	counter = 0
+	rospy.init_node('image_listener')
+
+	#topics for each leg
+
+	wheel1_topic = '/crobot/joint5_position_controller/state'
```
Comment (l1va) [inline tests/integration_test/scripts/sub.py:30]: copied Ramil's code.

File: tests/unit_test/unit.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                    [    0    ,    0     , 0, 1]])
+    return Rz
+def FK(q2):
+	a = np.array([[0],[0], [0], [1]])
+	P = Tz(0.9).dot(Tx(1.0).dot(Ty(-1.0).dot(Rz(q2).dot(Rz(90*3.142/180).dot(Ty(0.65).dot(Tz(-0.46).dot(a)))))))
+	return P[:3]
+class Test(unittest.TestCase):
+	def test_FK(self):
+		b = np.array([[0.38925516],[-1.22246514], [0.44]])
+		self.assertEqual(np.sum((FK(20*3.142/180)- b) < 0.001), 3)
```
Comment (l1va) [inline tests/unit_test/unit.py:37]: it is dangerous to test on sum but the interesting idea)

**Review by l1va — CHANGES_REQUESTED** [review]

File: my_robot_gazebo/launch/xacro.launch (hunk 1 above)
Comment (StalkerSanya) [reply my_robot_gazebo/launch/xacro.launch]: I checked something, but I have not remembered what already. In new comment "fixed" I  revealed

File: tests/integration_test/scripts/sub.py (hunk 2 above)
Comment (StalkerSanya) [reply tests/integration_test/scripts/sub.py]: Added asserts in new commit "fixed"

File: tests/integration_test/scripts/sub.py (hunk 3 above)
Comment (StalkerSanya) [reply tests/integration_test/scripts/sub.py:30]: Thanks for observation. Already fixed in new commit "fixed"

File: tests/unit_test/unit.py (hunk 4 above)
Comment (StalkerSanya) [reply tests/unit_test/unit.py:37]: It is dangerous, but it works (until)

**Review by l1va — APPROVED** [review]

**Pull request merged**